autoPyTorch/datasets/base_dataset.py

```diff
@@ -106,8 +106,14 @@ def __init__(
             val_transforms (Optional[torchvision.transforms.Compose]):
                 Additional Transforms to be applied to the validation/test data
         """
-        self.dataset_name = dataset_name if dataset_name is not None \
-            else hash_array_or_matrix(train_tensors[0])
+
+        # Explicit definition for mypy compatibility
+        self.dataset_name: str = ""
```

Comment:
Is there no other way to pass the mypy check here? 
Overall the variable declaration is not used at all and the variable is always reassigned. (Not sure how much it can be applied, but there can also be inline mypy check skips.)